dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/HTMLPageAssetRenderedAPIImpl.java

```diff
@@ -232,34 +249,44 @@ public String getPageHtml(
                 throws DotSecurityException, DotDataException {
 
         final Host host = this.hostWebAPI.getCurrentHost(request, context.getUser());
-        final IHTMLPage page = getHtmlPageAsset(context, host, request).getHTMLPage();
+        final HTMLPageUrl htmlPageUrl = getHtmlPageAsset(context, host, request);
+        final IHTMLPage page = htmlPageUrl.getHTMLPage();
 
         return new HTMLPageAssetRenderedBuilder()
                 .setHtmlPageAsset(page)
                 .setUser(context.getUser())
                 .setRequest(request)
                 .setResponse(response)
                 .setSite(host)
-                .getPageHTML();
+                .setURLMapper(htmlPageUrl.pageUrlMapper)
+                .setLive(htmlPageUrl.hasLive())
+                .getPageHTML(context.getPageMode());
     }
 
-    private HTMLPageUrl getHtmlPageAsset(
-            final PageContext context,
-            final Host host,
-            final HttpServletRequest request)
-                throws DotDataException, DotSecurityException {
+    private HTMLPageUrl getHtmlPageAsset(final PageContext context, final Host host, final HttpServletRequest request)
+            throws DotDataException, DotSecurityException {
 
         HTMLPageUrl htmlPageUrl = null;
-        IHTMLPage htmlPageAsset = findPageByContext(host, context);
+        for (final SearchPageFunction pageSearcher : pageSearchers) {
+            final Optional<HTMLPageUrl> optional = pageSearcher.search(context, host, request);
 
-        if (htmlPageAsset == null){
-            htmlPageUrl   = findByURLMap(context, host, request);
-            htmlPageAsset = getPageByUri(context.getPageMode(), host, htmlPageUrl.getPageUrl());
-        } else {
-            htmlPageUrl   = new HTMLPageUrl(htmlPageAsset);
+            if (optional.isPresent()) {
+                htmlPageUrl = optional.get();
+                break;
+            }
+        }
+
```

Comment:
Would a tertiary-like operator be cleaner here?  Something like:
```java
final HTMLPageUrl htmlPageUrl = findPageByContext(host, context)
   .orElse(findByURLMap(context, host, request))
   .orElse(null);
```

Reply:
done https://github.com/dotCMS/core/pull/17859/commits/0f3d1a924414cb79755fde511a0f7c60660d0bb9#diff-051d49b06feb0c7462b349b6b61988a8R251